src/cmake/file_api/toolchains.rs

```diff
@@ -0,0 +1,62 @@
+use std::convert::TryFrom;
```

Comment:
`Toolchains` is a [quite new addition to cmake](https://gitlab.kitware.com/cmake/cmake/-/merge_requests/5678) (merged < 7 months ago), and using an older `cmake` (my version was 3.19.1 - still quite recent!) causes the build to fail with a cryptic error. I had to update to 3.21.2 to make it work. I think as a minimum we need to check the cmake version and warn the user that she needs to upgrade it, if the version is too old. A bit like what we do when trying to detect a recent version of Python.

Reply:
Nice find. Yup, will do that.

Reply:
Another (maybe simpler) approach would be to try to work with older cmake versions as well. Do we really need the toolchain infos? Can't we just download and put _all_ toolchains on the $PATH? I think, due to the way Espressif creates their toolchains, there will be no problem with that approach as e.g. the `gcc` executable has a unique name across all toolchains.

Reply:
Just putting them on `PATH` wouldn't do much because we'd still have to know the path to the linker (so `gcc`). We need it for getting the sysroot for `bindgen` and for `ldproxy`.
You're right though we don't exactly need to use cmake to tell us where gcc is, we could also use the `which` crate to get gcc in `PATH`. But we need to have `cmake` >= 3.14 which is also pretty new (1-2 years depending on patch version). Only as of `esp-idf` v4.2 they switched to `cmake` version 3.16 in their `idf_tools.py` which I'm using in the build script.

Though we should really have this discussion on the build script PR not this one.